### ISSUE TITLE
no-duplicate-case incorrect error 

### ISSUE THREAD

**Issue by yuliapi** [eslint/eslint#12633]
These two cases give `Duplicate case label` error.
```
 case constants.TYPES.STATS_1_PROP_Z_TEST: {
    //do something
    break;
  }
  case constants.TYPES.STATS_2_PROP_Z_TEST: {
    //do something
    break;
  }

constants = {
  TYPES: {
       STATS_1_PROP_Z_TEST: 'STATS_1_PROP_Z_TEST',
       STATS_2_PROP_Z_TEST: 'STATS_2_PROP_Z_TEST',
  }
}
```

**Comment by mdjermanovic** [comment]
Hi @yuliapi, thanks for the bug report!

I couldn't reproduce the error using the code parts you provided. Here's an [online demo link](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tZHVwbGljYXRlLWNhc2U6IFwiZXJyb3JcIiovXG5cbnZhciBjb25zdGFudHMgPSB7XG4gIFRZUEVTOiB7XG4gICAgICAgU1RBVFNfMV9QUk9QX1pfVEVTVDogJ1NUQVRTXzFfUFJPUF9aX1RFU1QnLFxuICAgICAgIFNUQVRTXzJfUFJPUF9aX1RFU1Q6ICdTVEFUU18yX1BST1BfWl9URVNUJyxcbiAgfVxufSBcblxuc3dpdGNoIChmb28pIHtcbiAgICBjYXNlIGNvbnN0YW50cy5UWVBFUy5TVEFUU18xX1BST1BfWl9URVNUOiB7XG4gICAgICAgIC8vZG8gc29tZXRoaW5nXG4gICAgICAgIGJyZWFrO1xuICAgIH1cbiAgICBjYXNlIGNvbnN0YW50cy5UWVBFUy5TVEFUU18yX1BST1BfWl9URVNUOiB7XG4gICAgICAgIC8vZG8gc29tZXRoaW5nXG4gICAgICAgIGJyZWFrO1xuICAgIH1cbn1cblxuIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo1LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19) with the same two cases, but there is no `Duplicate case label` error.

Can you please fill out the [bug report template](https://github.com/eslint/eslint/blob/master/.github/ISSUE_TEMPLATE/BUG_REPORT.md) ?
